File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        mp[0] = 1;//ki sum-k=0 mtlb mill gaya ek subarray   
        int sum = 0;
        int count = 0;
        for (int a : nums) {
            sum=sum+a;
            if (mp.find(sum - k) != mp.end())//mtlb map mai element present h  
            {
                count=count+mp[sum - k];//value access kr dega
            }
            mp[sum]++;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count = 0;
        int sum = 0;
        Map<Integer, Integer> mp = new HashMap<>();
        mp.put(0, 1);
        for (int a = 0; a < nums.length; a++){
            sum += nums[a];
            if (mp.containsKey(sum - k)) {
                count += mp.get(sum - k);
            }
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }
        return count;
    }
}
